### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by JuanDBta**
The Ruby on Rails capstone project is about building a mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.

I created a Ruby on Rails application that allows the user to:

Register and log in, so that the data is private to them.
Introduce new transactions associated with a category.
See the money spent on each category.

I followed these design guidelines: (https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

Colors.
Typography: font face, size and weight.
Layout: composition and space between elements.
Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that I need to give appropriate credit to the author. Therefore, I did it in the README of the project.

Interactions:
- Splash screen
A simple page with the name of the app and links to the sign up and log in pages.

- Sign up and log in pages

The user is able to register in the app with full name, email and password.
The user can log into the app using email and password.

- Home page (categories page)
When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

- Add a new category" page

The user fills out a form to create a new category, indicating their name and icon.
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.

- "Add a new transaction" page

The user fills out a form to create a new transaction with:

Name
Amount
Category
The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.

- Created unit and integration tests for all the most important components of your RoR application.

- Technical requirements
I used Postgres database.
I used devise for authentication.
I validated all user input to make sure that anyone with bad intentions cannot compromise your app.

- The project is deployed and accessible online with Render.
- My database schema reflect the following structure:

erd_diagram

- Project documentation
I recorded a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video includes:

A description of the project.
A demo of the project features.
I also highlighted some interesting piece of the code I am very proud of.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          <% end %>
+          <li><h1 class="login-title">Add Category</h1></li>
+          <%= button_to 'Sign Out', destroy_user_session_path, class: "sign-out-button", method: :delete %>
+          <% else %>
+          <%= link_to(:back, class: "back") do %>
+            <i class="fas fa-arrow-left fa-3x"></i>
+          <% end %>
+            <li><h1 class="login-title">TRANSACTIONS</h1></li>
+            <%= button_to 'Sign Out', destroy_user_session_path, class: "sign-out-button", method: :delete %>
+          <% end %>
```
Comment (divinecharlotte) [inline app/views/layouts/application.html.erb:122]: - [ ] Kindly consider making sure that when a user is not logged they can't access pages that require to be logged in like : home, transaction, add a new category, and add new transaction to meet the project requirements

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @JuanDBta    , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- Clean code👍 
- Implemented some project requirements   👍 


## Required Changes ♻️

_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/views/layouts/application.html.erb:122]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @JuanDBta 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Github-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**